### ISSUE TITLE
setting number style for all currencies

### ISSUE THREAD

**Issue by markokocic** [simonmichael/hledger#142]
I have the following in the start of my transactions file:
`D 1,000.00 €`
But hledger prints only € amounts using this formatting, while using default formating for $. For example:

```
     -1000.00 $    income:category1
   -31,015.56 €    income:category2
```

 Right now it is not possible to specify global formatting options for all transactions.

**Comment by alerque** [comment]
Why use `D` at all, isn't [`commodity`](https://hledger.org/journal.html#declaring-commodities) the correct declaration for this use case?

**Comment by simonmichael** [comment]
Yes indeed, nowadays the commodity directive should be used for this (if needed). One directive per commodity - currently there's no way to set a default style to be used for all present and future commodities, it's [hard coded](https://github.com/simonmichael/hledger/blob/master/hledger-lib/Hledger/Data/Amount.hs#L151) (symbol on the left, no space, no digit group separators, period as decimal mark, no decimal places).

**Comment by alerque** [comment]
If anything then this is only a documentation issue, not a journal one. Defining anything like a default format to apply across multiple commodities seems like a can of worms we don't want to open. The current system of declaring each one or using the first seen format is better than anything that would clobber all of them.

**Comment by simonmichael** [comment]
What about the related (though separate) issue mentioned above - is it worth thinking about removing D's ability to set the display style, for simpler behaviour, at the cost of adding a slight Ledger incompatibility ? Or, let sleeping dogs lie ?

**Comment by alerque** [comment]
My inclination would be to throw a warning pointing to the recommended way to format commodities, but to leave the actual function alone both for Ledger and older Hledger file compatibility. I don't see any particular gain from stripping out the function, only in pointing users toward a more robust solution.
